Add support for global dashboards

GeorgianaElena opened the pull request:
This allows creating dashboards across all datasources.
It also adds an initial dashboards like the one below that shows weekly users across all clusters/datasources.
<img width="1346" alt="Screenshot 2022-07-04 at 18 20 04" src="https://user-images.githubusercontent.com/7579677/177183018-4c9f5aea-693d-43da-808c-6648a1f13299.png">

yuvipanda commented on deploy.py:
```diff
@@ -43,11 +43,25 @@ def ensure_folder(name, uid, api):
+
+    # If we need to build a global dashboard across more datasource
+    # pass the list of datasources to be used in the queries
+    if "global" in dashboard_path:
```
If I understand this correctly, we're not deploying some dashboards if we don't have more than one datasource - is that right?

GeorgianaElena replied:
Yes, we're not deploying the "global" ones.

So the set of rules would be:
- any dashboards that gather info across multiple datasources, should have the `global` keyword in their filenames
- any dashboards stored in files that contain the `global` keyword need the datasources to be passed as [`tla-code`](https://jsonnet.org/ref/language.html#passing-data-to-jsonnet)
- if a file has the `global` keyword, but we're not actually having more than one datasource, then the dashboards in that file don't get deployed at all.

The best place to document this currently is probably the README, right? What do you think @yuvipanda?

yuvipanda replied:
README seems great! Can we instead put it in a global/ directory instead of relying on the filename?

yuvipanda replied:
Actually, since deploy.py takes a 'directory' param, can we:

1. Move these to a 'global-dashboards' directory
2. Wherver they are to be deployed, just run deploy.py twice - once for the local dashboards and once for the global dashboards

This reduces any special casing. What do you think?

GeorgianaElena commented:
@yuvipanda, I believe I have addressed your feedback in the last couple of commits. Let me know if there's anything else that I can improve. Thanks! ✨

yuvipanda commented on deploy.py:
```diff
@@ -43,11 +43,28 @@ def ensure_folder(name, uid, api):
+
+    # If we need to build a global dashboard across more datasource
+    # pass the list of datasources to be used in the queries
+    if global_dash:
```
I'd love to completely get rid of this special casing! Sorry if I wasn't super clear in https://github.com/jupyterhub/grafana-dashboards/pull/39#discussion_r897847029. What I'd love to do is:

1. No specialcasing
2. Next to the 'dashboards/' directory, we have a `global-dashboards/` directory. By default, we only deploy the `dashboards/` directory
3. *If* on a particular grafana someone wants global-dashboards specifically, they can run `deploy.py` twice - once against `dashboards/` and once against `global-dashboards/` to get that. 

How does that sound?

GeorgianaElena replied:
Hmm, if I understand correctly, the global-dashboards will get installed only if this directory is explicity deployed by a human?

I'm not sure it's not possible without special casing in the deployer, because this "global" dashboard needs to be passed the list of datasources in the cmd like so:

```
datasources_names = [ds["name"] for ds in datasources]
cmd.extend(["--tla-code", f"datasources={datasources_names}"])
```

yuvipanda commented:
Thanks a lot, @GeorgianaElena!
